### ISSUE TITLE
(EKS): outputConfigCommand does not work any more without mastersRole

### ISSUE THREAD

**Issue by markusl** [aws/aws-cdk#32318]
### Describe the bug

`outputConfigCommand` does not output anything if `props.mastersRole` is not set.

### Regression Issue

- [X] Select this option if this issue appears to be a regression.

### Expected Behavior

Output config command

### Current Behavior

```ts
    const outputConfigCommand = (props.outputConfigCommand ?? true) && props.mastersRole;
    if (outputConfigCommand) {
      const postfix = commonCommandOptions.join(' ');
      new CfnOutput(this, 'ConfigCommand', { value: `${updateConfigCommandPrefix} ${postfix}` });
      new CfnOutput(this, 'GetTokenCommand', { value: `${getTokenCommandPrefix} ${postfix}` });
    }

```

### Reproduction Steps

Set `outputConfigCommand` to true

### Additional Information/Context

Seems to be broken in https://github.com/aws/aws-cdk/commit/51f0193bf34cca8254743561a1176e3ca5d83a74

### CDK CLI Version

2.171.0 (build 4957967)

### Node.js Version

20

### OS

Mac

### Language

TypeScript

**Comment by pahud** [comment]
This was an intentional breaking change from https://github.com/aws/aws-cdk/pull/25473

**Comment by markusl** [comment]
@pahud yes, it seems to be an intentional change, but I cannot see why. Please at least document the behavior and add a warning if the flag is set to true without mastersRole
